addons/account/tests/test_account_move_out_invoice.py

```diff
@@ -1319,6 +1319,51 @@ def test_out_invoice_line_onchange_analytic(self):
             },
         ], self.move_vals)
 
+    def test_out_invoice_analytic_default(self):
+        self.env.user.groups_id += self.env.ref('analytic.group_analytic_accounting')
```

Comment:
Bad news... your test is passing even without the diff above and the issue is still there...

Reply:
I fixed the issue on my side by replacing:
https://github.com/odoo/odoo/blob/master/addons/account/models/account_move.py#L1668
`line.partner = self.partner_id`
by
```
            if line.partner_id != self.partner_id:
                line.partner_id = self.partner_id
```